### DISCUSSION TITLE
No provider for SASL mechanism SCRAM-SHA-256: recompile librdkafka with libsasl2 or openssl support. Current build options: PLAIN

### DISCUSSION THREAD

**Discussion by kartikzheng** [fluent/fluent-bit#6086] [Q&A] [answered]
Hello everyone, I collect the log output to the authentication kafka, reported the following error

`[error] [output:kafka:kafka.0] failed to create producer: No provider for SASL mechanism SCRAM-SHA-256: recompile librdkafka with libsasl2 or openssl support. Current build options: PLAIN`

My configuration parameters are as follows:

` 

    rdkafka.request.required.acks 1

    rdkafka.security.protocol   sasl_plaintext

    rdkafka.sasl.username       XXXX

    rdkafka.sasl.password       XXXX

    rdkafka.sasl.mechanism      SCRAM-SHA-256
`

Is it the Librdkafka version? Do I need to recompile fluent-bit?

**Comment by kartikzheng** [comment, marked as answer]
> 

Install the following package and recompile fluent-bit to solve the problem.
yum -y install cyrus-sasl-gssapi cyrus-sasl-devel

**Comment by MicFlybird** [comment]
Hi. May I know the steps to recompile fluent-bit? thanks a lot.

**Reply by patrick-stephens** [reply]
Have a look at the packaging/build.sh script: this is what does it all for the official releases. We use a containerised build process to target whichever distro you're working with.
https://github.com/fluent/fluent-bit/blob/16eae107868bf03da2fce6ca4acd3f6ab2d3c4ff/packaging/build.sh

The setup for CentOS is here (I'm just guessing at the target @kartikzheng was using): https://github.com/fluent/fluent-bit/blob/master/packaging/distros/centos/Dockerfile
If those dependencies are added via a PR then this would be done for the official releases too so please contribute one if it helps.
